adding documentation around skipreceptornamescheck and new certificate validation in receptor (#593)
ansible/receptor: docs/source/tls.rst

Receptor supports mutual TLS authentication and encryption for above and below the
mesh connections.

> If you decide to consume TLS authentication in your mesh, the certificates OIDs (1.3.6.1.4.1.2312.19.1) will be verified against the `node.id` specified in the configuration file. If there is no match, the receptor binary will hard exit. You can `Skip Certificate Validation`_ as well.

Reviewer: This paragraph would fit nicely in the Above the mesh TLS section, as this is the first section that explains the concept of the above the mesh certification.

> If you decide to consume TLS authentication in your mesh, the certificates OIDs (1.3.6.1.4.1.2312.19.1) will be verified against the `node.id` you specify in the configuration file. If there isn't a match, the receptor binary will hard exit. You can `Skip Certificate Validation`_ as well.

Reviewer 2: ```suggestion
If you decide to consume TLS authentication in your mesh, the certificates OIDs (1.3.6.1.4.1.2312.19.1) will be verified against the `node.id` specified in the configuration file. If there is no match, the receptor binary will hard exit. You can `Skip Certificate Validation`_ as well.
```

SHA256 and SHA512 fingerprints are supported.  SHA1 fingerprints are not supported due to the insecurity of the SHA1 algorithm.

> Skip Certificate Validation

Reviewer: I suggest we move this section below `Above the mesh TLS`

^^^^^^^^^^^^^^^^^^^^^^^^^^^

Depending on the specifics of your environment(s), if you need to turn off certificate validation, you can add a `SkipReceptorNamesCheck` key-value pair in your configuration file for `tls-server`, `tls-config`, or both.

.. code-block:: yaml

    ---
    - node:
        id: bar

    - log-level:
        level: Debug

>     - tls-client:

Reviewer: I think for clarity we can just have a truncated config snippet here, with just the tls-client and tls-server portions. We do this elsewhere in the docs.

Author: i'd like to keep the `node.id` section in the code snippet, just to highlight the new relationship we're introducing between the certificates and the `node.id`. what do you think?

SHA256 and SHA512 fingerprints are supported.  SHA1 fingerprints are not supported due to the insecurity of the SHA1 algorithm.

Skip Certificate Validation
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you are in a development environment and need to turn off certificate validation, you can add a `SkipReceptorNamesCheck` key value pair in your configuration file for `tls-server`, `tls-config` or both.

Reviewer 2: ```suggestion
If you are in a development environment and need to turn off certificate validation, you can add a `SkipReceptorNamesCheck` key-value pair in your configuration file for `tls-server`, `tls-config`, or both.
```

> If you are in a development environment and need to turn off certificate validation, you can add a `SkipReceptorNamesCheck` key-value pair in your configuration file for `tls-server`, `tls-config`, or both.

Reviewer: there is a valid, production-level use case for having this set to True. If a user has separate certs for the TLS TCP backend connection and the control service, then having the receptor name in the TLS TCP cert is not required. If the receptor name is not present, the user must set `skipreceptornamescheck` to True for that cert definition.

Author: i made this block of code less ambiguous. please recheck @fosterseth !
